components/org.wso2.identity.event.common.publisher/src/main/java/org/wso2/identity/event/common/publisher/internal/EventPublisherDataHolder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.common.publisher.internal;
+
+import org.wso2.identity.event.common.publisher.EventPublisher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Event Publisher Data Holder.
+ */
+public class EventPublisherDataHolder {
+
+    private static final EventPublisherDataHolder instance = new EventPublisherDataHolder();
+    private List<EventPublisher> eventPublishers = new ArrayList<>();
+
+    private EventPublisherDataHolder() {
+
+    }
+
+    public static EventPublisherDataHolder getInstance() {
```

Comment:
Not synchronized ?

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/eb4cae044fe65111cb72917fd78af0619d056a02